akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala

```diff
@@ -567,7 +567,7 @@ final class ClusterMessageSerializer(val system: ExtendedActorSystem)
         member.getUpNumber,
         memberStatusFromInt(member.getStatus.getNumber),
         rolesFromProto(member.getRolesIndexesList.asScala.toSeq),
-        appVersionMapping(member.getAppVersionIndex))
+        if (appVersionMapping.isEmpty) Version.Zero else appVersionMapping(member.getAppVersionIndex))
```

Comment:
when receiving gossip from an old node (without app-version), the `appVersionMapping` will be empty. We fallback to Version.Zero in that case.